x/farm/keeper/plan.go

```diff
@@ -159,6 +164,10 @@ func (k Keeper) AllocateRewards(ctx sdk.Context) error {
 	if maxBlockDuration := k.GetMaxBlockDuration(ctx); blockDuration > maxBlockDuration {
 		blockDuration = maxBlockDuration
 	}
+	// If the block duration is 0, skip this block for rewards allocation.
+	if blockDuration == 0 {
```

Comment:
When can `blockDuration` be 0? As far as I see, `MaxBlockDuration` must be positive where 0 is not allowed in the first place.